### ISSUE TITLE
Fast R-CNN

### ISSUE THREAD

**Issue by Yagami360** [Yagami360/MachineLearning-Papers_Survey#75]
## 0. 論文情報・リンク
- 論文リンク：https://arxiv.org/abs/1504.08083
- 公開日時：2015/04/30
- 組織 : Microsoft Research
- 被引用数（記事作成時点）：10132 件
- 実装コード（Caffle）: https://github.com/rbgirshick/fast-rcnn
- Publication : 

## 1. どんなもの？
- R-CNN の処理速度を大幅に高速化し、物体検出精度も高めた深層畳み込みベースの物体検出モデル。現在多くの物体検出モデルやセグメンテーションモデルのベースラインとして採用されている。

## 2. 先行研究と比べてどこがすごいの？
- 既存の深層畳み込みベースの物体検出モデルである R-CNN は、（当時としては）優れた物体検出精度を実現したが、学習と推定速度ともに非常に低速であるという問題があった。
	本手法では、R-CNN を大幅に高速化し（学習時間：約１０倍以上、推定時間：約10０倍）、更に物体検出精度も高めた Fast R-CNN を提案している。
- PASCAL VOC 2012 での物体検出で、当時のSOTAを達成

## 3. 技術や手法の"キモ"はどこにある？
- 【R-CNN】前提知識
	R-CNN では、物体検出を以下の３つのモジュールに分けて行っている。
		a. 物体候補領域検出するモジュール (Region Proposal) 
		b. 得られた物体候補領域の特徴抽出を行うモジュール（特徴抽出は CNNで行う）
		c. 得られた物体候補領域から分類を行うモジュール(分類はSVMで行う) 
	
	物体候補領域検出モジュールで検出された候補領域は複数存在するが、R-CNN はこれら複数候補領域”全て”に対して、CNN で特徴抽出し、SVMで分類を行っている。そのため、非常に低速であるという問題が存在する。
	Fast R-CNN では、物体候補領域ではなく元の入力画像に対して CNN で特徴抽出することで、複数回 CNN を通さなくてよくして、処理速度を大幅に高速化するというのが基本的なコンセプトである

- アーキテクチャの全体像
    <image src="https://user-images.githubusercontent.com/25688193/86513342-692f9100-be44-11ea-9c1d-7e367d8b064a.png" width="492">
	上図は、Fast R-CNN のアーキテクチャ全体像を示した図である。
	このアーキテクチャでの処理は、以下のような流れになる。
	
	1. 入力画像全体を ImageNet で事前学習済み CNN モデル（VGG等）に入力し、入力画像全体に対しての特徴マップを生成する。
		
	1. RoI領域（上図赤枠）を上記 ImageNet で事前学習済み CNN に入力し、上記入力画像全体に対しての特徴マップ上へ射影する
		※ RoI領域は、CNN ではなくて、Selective Search（SS）で行っている。
		※ 後発の Faster-RCNN では、この RoI 領域の検出も CNN で行うことで、処理速度と精度を高めている。
		
	1. 特徴マップ上へ射影された RoI 領域を、RoI Pooling layer へ入力し、固定長の特徴ベクトルを抽出
		
	1. 抽出した特徴ベクトルを全畳み込み層（FC層）へ入力し、softmax 出力と BBOX のオフセット値出力に変換
		※ sofmax 出力は、物体カテゴリ数 K+1 個のベクトルで、k 番目の要素が物体カテゴリ k である確率を表す
		※ BBOX のオフセット値出力は、物体カテゴリ k の座標位置 (x,y) と長方形サイズ (w,h) のペアデータ

	これらのネットワーク全体は、ImageNet で事前学習済み CNN モデル（VGG等）を fine-tuning したネットワークで構成され、end2end に学習される。

- RoI Pooling layer
    <image src="https://user-images.githubusercontent.com/25688193/86513374-adbb2c80-be44-11ea-81d1-90af1aa6077f.png" width="350">
	RoI Pooling layer では、上図のように、ます入力画像全体の特徴マップ上に射影された RoI 領域を、４つのグリッドに分割する。その後、それぞれのグリッドに対して max pooling 演算を実施し、固定解像度（例 : 7x7）のより小さな特徴マップに変換する。

- ImageNet で事前学習済み CNN モデルの fine-tuning
	先に述べたように、このネットワーク全体は、ImageNet で事前学習済み CNN モデル（VGG 等）を fine-tuning したネットワークで構成されている。
	具体的な fine-tuning 内容は、以下のようになる。
	
	1. RoI Pooling layer での fine-tuning
		RoI Pooling layer での pooling 層は、ImageNet で事前学習済み CNN モデルの最後の層での pooling 層を利用している。但し、入出力サイズは互換性を持たせるため fine-tuning されている。
		
	2. softmax 出力層と BBOX のオフセット値出力層
		事前学習済みCNNモデルの出力層を、softmax 出力層と BBOX のオフセット値出力層に fine-tuning
		
	3. ネットワークの入力
		画像のリストとその画像の RoI リストを入力出来るように、事前学習済みCNNモデルの入力層を fine-tuning

- 損失関数（Multi-task loss）
    本手法でのネットワークは、以下の式で定義される Multi-task loss によって、end2end に学習される。
    <image src="https://user-images.githubusercontent.com/25688193/86513394-d4796300-be44-11ea-8d25-531402492c94.png" width="770">

- SVD（特異値分解）によるFC層の高速化
	Fast R-CNN では、CNN による畳み込み処理の回数を減らして高速化した結果、全体の処理時間に対する全畳み込み層（FC層）の割合が大きくなってしまい、この部分が処理時間の面でボトルネックになるようになった。
	そのため、このFC層の高速化も行っている。
	具体的には、以下の式のように、全畳み込み層での行列計算をSVD（特異値分解）で近似するようにする。
    <image src="https://user-images.githubusercontent.com/25688193/86513402-ea872380-be44-11ea-9b6c-2a1b2bfcae82.png" width="700">

## 4. どうやって有効だと検証した？
- 既存の物体検出モデルとの物体検出精度の定量比較検証
    <image src="https://user-images.githubusercontent.com/25688193/86513419-05599800-be45-11ea-9bca-8f70adc1cb1e.png" width="630">
    Pascal VOC データセットにおいて、既存の物体検出モデルと比較して、ほとんど全ての物体検出精度スコアで上回っている。

- 既存の物体検出モデルとの処理速度の定量比較検証
    <image src="https://user-images.githubusercontent.com/25688193/86513436-2f12bf00-be45-11ea-9138-9f1eb7040cca.png" width="385">
    R-CNN, SPPNet と比較して、大幅に高速化している。（学習時間：約１０倍以上、推定時間：約10０倍）

## 5. 議論はあるか？
- xxx

## 6. 次に読むべき論文はあるか？
- Faster R-CNN

## 7. 参考文献
- [論文紹介: Fast R-CNN&Faster R-CNN](https://www.slideshare.net/takashiabe338/fast-rcnnfaster-rcnn)
